### PR TITLE
[DO NOT MERGE] Made it so that the Tables clients don't raise an exception when deleting a table/entity that does not exist.

### LINKED ISSUE #14490: [Tables] Don't raise an exception when deleting a table/entity that does not exist
Architects would like the delete method to swallow the 404 that occurs if the table/entity trying to be deleted is not found.

- fix deleteEntity method to not throw a 404: entity not found
- fix deleteTable method to not throw a 404: table not found

### PR DESCRIPTION
**Pull request by vcolin7**
Fixes #14490.

Note: I was wondering if it would be better to move the method I added to `TablesUtils` to somewhere like Azure Core so it can be used on more SDKs. I can see that sometimes we might want to swallow and exception based on certain response status codes, but I'm not really sure how common this is. Any thoughts @alzimmermsft, @anuchandy, @srnagar?

### CODE REVIEW

**Comment by christothes** [comment]
I don't think that we made this behavior change cross language. @annatisch Do you recall if we decided not to do this or if we should all make this change?

One issue I recall is that a change like this would not provide any feedback as to whether the table was actually deleted or if it didn't exist. I believe where we landed was to prefer a method like `DeleteIfExists` instead - but I had that as a low priority item in the .NET backlog.

**Comment by annatisch** [comment]
Yeah this would currently raise a ResourceNotExists error in Python. I think we had this discussion regarding deleteTables, but not sure about deleteEntity. I would expect those two to behave the same way - In Java do your deleteResource operations usually not raise in the case that resource isn't found?

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TablesUtils.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+     * @param headersClass Class of the headers object to deserialize from the HTTP response.
+     * @param serializerAdapter Serializer adapter to deserialize the given HTTP response.
+     * @param logger Logger that will be used to record the exception.
+     * @param <E> Response exception class.
+     * @param <R> Deserialized response class.
+     * @param <H> Deserialized headers class.
+     * @return A Mono that contains the deserialized response.
+     */
+    static <E extends HttpResponseException, R extends ResponseBase<H, Void>, H> Mono<R> swallowExceptionForStatusCode(
+        int statusCode, E httpResponseException, Class<R> responseClass, Class<H> headersClass,
```
Comment (alzimmermsft) [inline sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TablesUtils.java:40]: I would make this Tables specific so we don't need to handle serialization/deserialization logic here. Additionally, `HttpResponseException` has a property for `HttpHeaders`, and looking at the APIs calling into this they simply return `Mono<Response<Void>>`/`Mono<Void>` so we don't need to play around with the POJO `HttpHeaders` representation from swagger generation.

**Comment by vcolin7** [comment]
@annatisch @christothes The issue this is based on is a few months old and I don't know exactly the context this feedback was given to us: Was it during an arch board meeting for all languages? Was it just for Java? To me it sounds like the former, but I could be wrong.

I don't think our Java operations ignore 404 responses in general, so maybe we could just move this to a method like `deleteIfExists` like Chris suggested.

**Comment by vcolin7** [comment]
Closing in favor of doing this in an updated PR.

**Pull request closed without merging**